x/assets/keeper/genesis.go

```diff
@@ -0,0 +1,59 @@
+package keeper
+
+import (
+	"github.com/ExocoreNetwork/exocore/x/assets/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// InitGenesis initializes the module's state from a provided genesis state.
+func (k Keeper) InitGenesis(ctx sdk.Context, data *types.GenesisState) {
+	if err := k.SetParams(ctx, &data.Params); err != nil {
+		panic(err)
+	}
+	// client_chain.go
```

Comment:
Each time the loop iterates, the infoCopy variable is assigned to the info variable, and this assignment operation actually creates a copy of each element in the data.ClientChains or data.Tokens arrays. If these arrays contain a large number of elements, or if the elements themselves take up a large amount of memory, then this copy operation can result in significant memory overhead